docs/sphinx-docs/build_sphinx.py

```diff
@@ -176,25 +177,36 @@ def apidoc():
     # Clean directory before generating a new version.
     #_remove_dir(SASVIEW_API_TARGET)
 
-    subprocess.check_call([
-        "sphinx-apidoc",
-        "-o", SASVIEW_API_TARGET, # Output dir.
-        "-d", "8", # Max depth of TOC.
-        "-H", "SasView", # Package header
-        SASVIEW_BUILD,
-        # omit the following documents from the API documentation
-        joinpath(SASVIEW_BUILD, "sas", "qtgui", "convertUI.py")
-    ])
+    try:
+        subprocess.check_output([
+            "sphinx-apidoc",
+            "-o", SASVIEW_API_TARGET, # Output dir.
+            "-d", "8", # Max depth of TOC.
+            "-H", "SasView", # Package header
+            SASVIEW_BUILD,
+            # omit the following documents from the API documentation
+            joinpath(SASVIEW_BUILD, "sas", "qtgui", "convertUI.py")
+        ])
+
+    except CalledProcessError as e:
+        print(e.output)
+        raise e
+
+    try:
+        subprocess.check_output([
+            "sphinx-apidoc",
+            "-o", SASMODELS_API_TARGET, # Output dir.
+            "-d", "8", # Max depth of TOC.
+            "-H", "sasmodels", # Package header
+            SASMODELS_BUILD,
+            # omit the following documents from the API documentation
+            joinpath(SASMODELS_BUILD, "sasmodels", "models"),
+        ])
+
+    except CalledProcessError as e:
+        print(e.output)
```

Comment:
Any error is likely to be on stderr, which is not captured by `check_output()`. Lots of sharp edges to cut yourself on in the `subprocess` module.

The `check_output` could just become a `run` call (since that is now recommended anyway), ensuring that both stdout and stderr are captured. For `run`, either `check=True` will give the same exception on nonzero exit status, or check the `CompletedProcess` object rather than use exceptions.

https://docs.python.org/3/library/subprocess.html#subprocess.run

Reply:
I'm not sure whether sphinx considers that a supported interface — I know they're busy removing setuptools integration, for instance.